pkg/kubectl/cmd/config/create_authinfo.go

```diff
@@ -35,20 +36,34 @@ type createAuthInfoOptions struct {
 	clientCertificate util.StringFlag
 	clientKey         util.StringFlag
 	token             util.StringFlag
+	username          util.StringFlag
+	password          util.StringFlag
 }
 
 func NewCmdConfigSetAuthInfo(out io.Writer, pathOptions *pathOptions) *cobra.Command {
 	options := &createAuthInfoOptions{pathOptions: pathOptions}
 
 	cmd := &cobra.Command{
-		Use:   fmt.Sprintf("set-credentials name [--%v=path/to/auth/file] [--%v=path/to/certficate/file] [--%v=path/to/key/file] [--%v=bearer_token_string]", clientcmd.FlagAuthPath, clientcmd.FlagCertFile, clientcmd.FlagKeyFile, clientcmd.FlagBearerToken),
+		Use:   fmt.Sprintf("set-credentials name [--%v=path/to/auth/file] [--%v=certfile] [--%v=keyfile] [--%v=bearer_token] [--%v=basic_user] [--%v=basic_password]", clientcmd.FlagAuthPath, clientcmd.FlagCertFile, clientcmd.FlagKeyFile, clientcmd.FlagBearerToken, clientcmd.FlagUsername, clientcmd.FlagPassword),
 		Short: "Sets a user entry in .kubeconfig",
-		Long: `Sets a user entry in .kubeconfig
-	Specifying a name that already exists will merge new fields on top of existing values for those fields.
-	e.g. 
-		kubectl config set-credentials cluster-admin --client-key=~/.kube/cluster-admin/.kubecfg.key
-		only sets the client-key field on the cluster-admin user entry without touching other values.
-		`,
+		Long: fmt.Sprintf(`Sets a user entry in .kubeconfig
+
+  Specifying a name that already exists will merge new fields on top of existing
+  values. For example, the following only sets the "client-key" field on the 
+  "cluster-admin" entry, without touching other values:
+
+    set-credentials cluster-admin --client-key=~/.kube/admin.key
+
+  Only one type of credentials can be set per user:
```

Comment:
Bearer token and basic auth are mutually exclusive, but is client cert supposed to be mutually exclusive with them as well? The current auth scheme used by kubectl includes both a client cert and basic auth; I'm pretty sure it fails if either is missing.